[javascript] Update Rhino library to 1.7.13 (#3242)
pmd/pmd: docs/pages/release_notes.md

### New and noteworthy

#### Javascript module now requires at least Java 8

The latest version of [Rhino](https://github.com/mozilla/rhino), the implementation of JavaScript we use
for parsing JavaScript code, requires at least Java 8. Therefore we decided to upgrade the pmd-javascript

> module to Java 8 as well. This means, that from now on, a Java 8 or later runtime is required in order

Reviewer: ```suggestion
module to Java 8 as well. This means that from now on, a Java 8 or later runtime is required in order
```
I like German commas, but the sentence would breathe better without this one ;)

> to analyze JavaScript code. Note, that PMD core still stays the at Java 7.

Reviewer: ```suggestion
to analyze JavaScript code. Note that PMD core still only requires Java 7.
```
